### PR TITLE
fix(docs): use concrete secrets from settings

### PR DESCRIPTION
**Pull request by sthulb**
**Issue number:** #2277

### Changes

> Please provide a summary of what's being changed

Forces GHA to use the secrets stored in the repository.

### User experience

> Please share what the user experience looks like before and after this change

No change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [X] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [X] I have performed a self-review of this change
* [X] Changes have been tested
* [X] Changes are documented
* [X] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Review by heitorlessa — CHANGES_REQUESTED** [review]
use GitHub Environments secrets instead (encrypted vs unencrypted)

File: .github/workflows/reusable_publish_docs.yml [hunk 1]
```diff
@@ -5,6 +5,13 @@ env:
 
 on:
   workflow_call:
+    secrets:
+      AWS_DOCS_ROLE_ARN:
+        description: Docs IAM Role
+        required: false
+      AWS_DOCS_BUCKET:
+        description: Docs S3 Bucket
+        required: false
```
Comment (heitorlessa) [inline .github/workflows/reusable_publish_docs.yml]: use `secrets: inherit` instead from the caller

Comment (rubenfonseca) [reply .github/workflows/reusable_publish_docs.yml]: In addition to this, use `environment: Docs` in the caller, and move the secrets into the `Docs` environment please. Don't leave any "Repository secrets".

**Review by rubenfonseca — CHANGES_REQUESTED** [review]

**Review by leandrodamascena — APPROVED** [review]
APPROVED!

**Pull request merged**